server/container/guice/memory-guice/src/test/java/org/apache/james/GuiceJamesServerTest.java

```diff
@@ -99,5 +197,24 @@ void serverShouldNotBeStartedOnUncaughtException(GuiceJamesServer server) throws
 
             assertThat(server.isStarted()).isFalse();
         }
+
+        @Test
+        void serverShouldThrowWhenAStartUpCheckFails(GuiceJamesServer server) throws Exception {
```

Comment:
Can we also assert that the server is not started in the event of a failed startup check?